rs/replicated_state/src/canister_state/system_state.rs

```diff
@@ -376,11 +384,15 @@ impl TaskQueue {
     }
 
     pub fn is_empty(&self) -> bool {
-        self.paused_or_aborted_task.is_none() && self.queue.is_empty()
+        self.paused_or_aborted_task.is_none()
+            && !self.on_low_wasm_memory_hook_status.is_ready()
+            && self.queue.is_empty()
     }
 
     pub fn len(&self) -> usize {
-        self.paused_or_aborted_task.as_ref().map_or(0, |_| 1) + self.queue.len()
+        self.paused_or_aborted_task.is_some() as usize
+            + self.on_low_wasm_memory_hook_status.is_ready() as usize
```

Comment:
I prefer the more explicit approach we had before and avoiding tricks like casting bools/options to a `usize`.